### PR TITLE
MKR regression tests: cat-exhaustiveness

### PR DESCRIPTION
**Pull request by ehildenb**
This is a proof that the Cat contract always correctly reverts on invalid function encoding inputs.

### CODE REVIEW

File: tests/specs/mcd/verification.k [hunk 1]
```diff
@@ -28,4 +28,10 @@ module VERIFICATION
 
     rule hash2(V1, _) ==K hash2(V2, _) => false requires V1 =/=Int V2
 
+    // ### cat-exhaustiveness
+
+    rule #sizeByteArray(_WS [ START .. WIDTH ]) => WIDTH requires 0 <=Int START andBool 0 <=Int WIDTH
```
Comment (denis-bogdanas) [inline tests/specs/mcd/verification.k:33]: `andBool START +Int WIDTH <=Int #sizeByteArray(WS)` , with `_WS` renamed to `WS`. In case it matters.

Comment (ehildenb) [reply tests/specs/mcd/verification.k:33]: I don't think it does matter does it? `_[_.._]` pads with zeros if `START +Int WIDTH >Int #sizeByteArray(_)`, so I think this lemma is correct either way, and the simpler side condition will perform better. Can you say why you think the extra side condition is needed?

Comment (ehildenb) [reply tests/specs/mcd/verification.k:33]: In fact, with the additional side-condition, this proof then fails to simplify some terms, and so it cannot be proven.

Comment (denis-bogdanas) [reply tests/specs/mcd/verification.k:33]: Forgot about padding. Then it's all good.

**Pull request merged**